ios/objc/apis/TorchTensor.h

```diff
@@ -0,0 +1,102 @@
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NS_ENUM(NSUInteger, TorchTensorType) {
+    TorchTensorTypeByte, //8bit unsigned integer
+    TorchTensorTypeInt,  //32bit signed integer
+    TorchTensorTypeLong, //64bit signed integer
+    TorchTensorTypeFloat, //32bit single precision floating point
+    TorchTensorTypeUndefined, //Undefined tensor type. This indicates an error with the model
+};
+
+@interface TorchTensor : NSObject<NSCopying>
+/**
+ Data type of the tensor
+ */
+@property(nonatomic,assign, readonly) TorchTensorType type;
```

Comment:
call it `.dtype`

Reply:
`.dataType` then?
Because such shortenings are against Apple's API guidelines for ObjC.